fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java

```diff
@@ -76,11 +74,9 @@ public Connection getConnection() throws SQLException {
 
     @Override
     public List<String> listDbNames() {
-        try (Connection connection = getConnection()) {
-            return Lists.newArrayList(schemaResolver.listSchemas(connection));
-        } catch (SQLException e) {
-            throw new StarRocksConnectorException(e.getMessage());
-        }
+        ImmutableMap<String, String> metaInfo =
+                ImmutableMap.of("func", "listDbNames");
+        return catalogCache.get(metaInfo, k -> metaResolver.refreshCacheForListDbNames());
```

Comment:
as async cache, if it doesn't hit on cache, it will return null? if you need wait the result why not use sync cache?

Reply:
The asynchrony here does not mean that the thread has returned, but rather that the thread is suspended and waiting for the result to return. For high concurrency databases, this is a good way.

Reply:
For more information, please refer to : https://zhuanlan.zhihu.com/p/611934066